dune/actions/storm.py

```diff
@@ -29,29 +30,34 @@ def get_faction_order(game_state):
     return faction_order
 
 
-class Storm(Action):
-    name = "storm"
-    ck_round = "storm"
-    su = True
+def do_storm_round(game_state, advance):
+    game_state.storm_position = (game_state.storm_position + advance) % 18
 
-    def _execute(self, game_state):
-        new_game_state = deepcopy(game_state)
+    destroy_in_path(game_state,
+                    range(game_state.storm_position, game_state.storm_position + 1))
 
-        advance = new_game_state.storm_deck.pop(0)
+    game_state.round = "spice"
 
-        new_game_state.storm_position = (new_game_state.storm_position + advance) % 18
+    game_state.ornithopters = []
+    carthag = game_state.map_state["Carthag"]
+    arrakeen = game_state.map_state["Arrakeen"]
+    if carthag.forces:
+        game_state.ornithopters.append(list(carthag.forces.keys())[0])
+    if arrakeen.forces:
+        game_state.ornithopters.append(list(arrakeen.forces.keys())[0])
 
-        destroy_in_path(new_game_state,
-                        range(game_state.storm_position, new_game_state.storm_position + 1))
 
-        new_game_state.round = "spice"
+class Storm(Action):
+    name = "storm"
+    ck_round = "storm"
+    su = True
 
-        new_game_state.ornithopters = []
-        carthag = new_game_state.map_state["Carthag"]
-        arrakeen = new_game_state.map_state["Arrakeen"]
-        if carthag.forces:
-            new_game_state.ornithopters.append(list(carthag.forces.keys())[0])
-        if arrakeen.forces:
-            new_game_state.ornithopters.append(list(arrakeen.forces.keys())[0])
+    @classmethod
+    def _check(cls, game_state, faction):
+        if any("Weather-Control" in state.treachery for state in game_state.faction_state.values()):
```

Comment:
This admits a "timing attack" to determine if someone in the game currently has weather control... I'm not sure if this matters or is inevitable without adding some kind of time out which is currently not in the game. I added in the explicit "pass" in other places even if you don't have the card so that you don't have to admit that you don't have the card.

Reply:
Oh right.  I'll fix this - just working on the old hajr now.